django/cantusdb_project/users/tests.py

```diff
@@ -29,6 +32,107 @@ def test_view(self):
         self.assertEqual(len(response.context["users"]), 6)
 
 
+class IndexerListViewTest(TestCase):
+    def setUp(self):
+        # unless a segment is specified when a source is created, the source is automatically assigned
+        # to the segment with the name "CANTUS Database" - to prevent errors, we must make sure that
+        # such a segment exists
+        Segment.objects.create(name="CANTUS Database")
+
+    def test_view_url_path(self):
+        response = self.client.get("/indexers/")
+        self.assertEqual(response.status_code, 200)
+
+    def test_view_url_reverse_name(self):
+        response = self.client.get(reverse("indexer-list"))
+        self.assertEqual(response.status_code, 200)
+
+    def test_url_and_templates(self):
+        """Test the url and templates used"""
+        response = self.client.get(reverse("indexer-list"))
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, "base.html")
+        self.assertTemplateUsed(response, "indexer_list.html")
+
+    def test_only_public_indexer_visible(self):
+        """In the indexer list view, only public indexers (those who have at least one published source) should be visible"""
```

Comment:
this is not the way I would have expected things to work, but okay.

Reply:
This is not ideal for me either, but it is how it works on old Cantus